app/src/main/kotlin/at/bitfire/davdroid/ui/DebugInfoActivity.kt

```diff
@@ -120,101 +166,216 @@ class DebugInfoActivity : AppCompatActivity() {
-                startActivity(Intent.createChooser(intent, null))
-            }
-            binding.causeView.setOnClickListener(showDebugInfo)
-            binding.debugInfoView.setOnClickListener(showDebugInfo)
+            builder.intent.addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION or Intent.FLAG_ACTIVITY_NEW_TASK)
+            builder.startChooser()
 
-            binding.fab.apply {
-                setOnClickListener { shareArchive() }
-                isEnabled = true
-            }
-            binding.zipShare.setOnClickListener { shareArchive() }
+            // Not beautiful, because it changes model data from the view.
+            // See https://github.com/android/architecture-components-samples/issues/63
+            model.zipFile.value = null
         }
 
-        model.logFile.observe(this) { logs ->
-            binding.logsView.setOnClickListener {
-                val uri = FileProvider.getUriForFile(this, getString(R.string.authority_debug_provider), logs)
-                val intent = Intent(Intent.ACTION_VIEW)
-                intent.setDataAndType(uri, "text/plain")
-                intent.addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION)
-                startActivity(Intent.createChooser(intent, null))
-            }
-        }
+        setContent { 
+            MdcTheme {
+                val debugInfo by model.debugInfo.observeAsState()
+                val zipProgress by model.zipProgress.observeAsState(false)
 
-        model.zipFile.observe(this) { zipFile ->
-            if (zipFile != null) {
-                // ZIP file is ready
-                val builder = ShareCompat.IntentBuilder(this)
-                    .setSubject("${getString(R.string.app_name)} ${BuildConfig.VERSION_NAME} debug info")
-                    .setText(getString(R.string.debug_info_attached))
-                    .setType("*/*")     // application/zip won't show all apps that can manage binary files, like ShareViaHttp
-                    .setStream(
-                        FileProvider.getUriForFile(
-                            this,
-                            getString(R.string.authority_debug_provider),
-                            zipFile
-                        )
+                val snackbarHostState = remember { SnackbarHostState() }
+
+                Scaffold(
+                    floatingActionButton = {
+                        if (debugInfo != null && !zipProgress) {
+                            FloatingActionButton(
+                                onClick = model::generateZip
+                            ) {
+                                Icon(Icons.Rounded.Share, stringResource(R.string.share))
+                            }
+                        }
+                    },
+                    scaffoldState = ScaffoldState(
+                        drawerState = remember { DrawerState(DrawerValue.Closed) },
+                        snackbarHostState = snackbarHostState
                     )
```

Comment:
When we would use `snackbarHost =` as [described here](https://developer.android.com/jetpack/compose/components/snackbar#basic_example), we would get rid of the `DrawerState` (and we don't have a drawer).